### PR TITLE
Upgrade ydb-go-sdk + fix usages + fix linter issues + add github acti…

### PR DESCRIPTION
**Pull request by asmyasnikov**
…on with integration tests with docker + add issue templates

### CODE REVIEW

File: storage/spanstore/reader/reader.go [hunk 1]
```diff
@@ -414,10 +420,11 @@ func (s *SpanReader) spansFromPartition(ctx context.Context, traceID model.Trace
 		if err != nil {
 			return err
 		}
-		defer res.Close()
```
Comment (apilikov) [inline storage/spanstore/reader/reader.go:417]: Could you please clarify the purpose of this diff?

**Review by apilikov — APPROVED** [review]

File: storage/spanstore/reader/reader.go [hunk 2]
```diff
@@ -316,17 +320,19 @@ func (s *SpanReader) queryPartitionList(ctx context.Context) ([]schema.Partition
...
-					return err
-				}
-				result = append(result, part)
+		defer func() {
+			_ = res.Close()
+		}()
+		if !res.NextResultSet(ctx, "part_date", "part_num", "is_active") {
+			return ErrNothingResultSets
+		}
+		result = make([]schema.PartitionKey, 0, res.ResultSetCount())
```
Comment (holycheater) [inline storage/spanstore/reader/reader.go:329]: this slice size should be number of rows, not number of result sets

File: storage/spanstore/reader/reader.go [hunk 3]
```diff
@@ -443,23 +450,27 @@ func (s *SpanReader) spansFromPartition(ctx context.Context, traceID model.Trace
...
-					table.ValueParam("$offset", types.Uint64Value(offset)),
-				),
-			)
-			if err != nil {
-				return err
-			}
-			defer res.Close()
-
-			for res.NextResultSet(ctx, "trace_id_low", "trace_id_high", "span_id", "operation_name", "flags", "start_time", "duration", "extra") {
+			if err = func() error { // for auto-call defer per each i
```
Comment (holycheater) [inline storage/spanstore/reader/reader.go:453]: This construct is barely readable
```go
err := func() error { /* ... */ }
if err != nil {
 // .. stuff
}
```

File: storage/spanstore/reader/reader.go [hunk 4]
```diff
@@ -658,20 +671,22 @@ func (s *SpanReader) execQuery(ctx context.Context, span opentracing.Span, query
...
-					return fmt.Errorf("scan failed: %w", err)
-				}
-				result = append(result, qr)
+		defer func() {
+			_ = res.Close()
+		}()
+		if !res.NextResultSet(ctx, "trace_ids", "rev_start_time") {
+			return ErrNothingResultSets
+		}
+		result = make([]dbmodel.IndexResult, 0, res.ResultSetCount())
```
Comment (holycheater) [inline storage/spanstore/reader/reader.go:680]: same, wrong count

**Review by holycheater — CHANGES_REQUESTED** [review]

File: storage/spanstore/reader/reader.go (hunk 3 above)
Comment (asmyasnikov) [reply storage/spanstore/reader/reader.go:453]: ack

File: storage/spanstore/reader/reader.go (hunk 1 above)
Comment (asmyasnikov) [reply storage/spanstore/reader/reader.go:417]: some linters (errcheck for example) required check error result on res.Close()

File: storage/spanstore/reader/reader.go (hunk 2 above)
Comment (asmyasnikov) [reply storage/spanstore/reader/reader.go:329]: Ack

File: storage/spanstore/reader/reader.go (hunk 4 above)
Comment (asmyasnikov) [reply storage/spanstore/reader/reader.go:680]: ack

**Review by holycheater — APPROVED** [review]

**Pull request merged**